src/mame/video/starshp1.cpp

```diff
@@ -354,7 +354,8 @@ uint32_t starshp1_state::screen_update_starshp1(screen_device &screen, bitmap_in
 	if (m_circle_kill == 0 && m_circle_mod == 0)
 		draw_circle(bitmap);
 
-	m_bg_tilemap->draw(screen, bitmap, cliprect, 0, 0);
+	rectangle tilemaprect(0, m_bg_tilemap->width(), 0, m_bg_tilemap->height());
```

Comment:
Do you want to calculate the intersection with the clipping rectangle here?  It will be more efficient with partial updates.

Reply:
I'm not sure why any calculation is needed.  It is a 32x8 grid of chars which is only drawn on the top 1/4 of the screen, fixed by the hardware.  I was surprised that, by default, it gets repeated 4x across the entire screen unless you manually set the cliprect like this.  Why does a small tilemap fill the screen by default?

Reply:
You probably want to calculate the intersection so you don’t do unnecesary work on partial updates.

Tilemaps are drawn so they fill the clipping rectangle you supply, wrapping if necessary.  This is pretty normal behaviour for tilemap hardware, because binary counters loop around to zero when they overflow.  Many games depend on this with a RAM-based tilemap slightly larger than the screen to implement a scrolling background.

Reply:
You do it like this before the draw() call:
tilemaprect &= cliprect;

Reply:
Ah, I get it now - thanks!